plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1262 @@
+	Pool DBPoolStatistics `json:"pool"`
+}
+
+// DBPoolStatistics pool related statistics
+type DBPoolStatistics struct {
+	ActiveCount    interface{} `json:"ActiveCount"`
+	AvailableCount interface{} `json:"AvailableCount"`
+	InUseCount     interface{} `json:"InUseCount"`
+}
+
+// JVMResponse GetJVMStat expected response type
+type JVMResponse struct {
+	Outcome string     `json:"outcome"`
+	Result  JVMMetrics `json:"result"`
+}
+
+// JVMMetrics JVM related metrics type
+type JVMMetrics struct {
+	Type map[string]interface{} `json:"type"`
+}
+
+// WebResponse getWebStatistics expected response type
+type WebResponse struct {
+	Outcome string                 `json:"outcome"`
+	Result  map[string]interface{} `json:"result"`
+}
+
+// DeploymentResponse GetDeployments expected response type
+type DeploymentResponse struct {
+	Outcome string            `json:"outcome"`
+	Result  DeploymentMetrics `json:"result"`
+}
+
+// DeploymentMetrics deployment related type
+type DeploymentMetrics struct {
+	Name          string                 `json:"name"`
+	RuntimeName   string                 `json:"runtime-name"`
+	Status        string                 `json:"status"`
+	Subdeployment map[string]interface{} `json:"subdeployment"`
+	Subsystem     map[string]interface{} `json:"subsystem"`
+}
+
+// WebMetrics  Web Modules related metrics
+type WebMetrics struct {
+	ActiveSessions    string                 `json:"active-sessions"`
+	ContextRoot       string                 `json:"context-root"`
+	ExpiredSessions   string                 `json:"expired-sessions"`
+	MaxActiveSessions string                 `json:"max-active-sessions"`
+	SessionsCreated   string                 `json:"sessions-created"`
+	Servlet           map[string]interface{} `json:"servlet"`
+}
+
+// JBoss the main collectod struct
+type JBoss struct {
+	Servers []string `toml:"servers"`
+	Metrics []string `toml:"metrics"`
+
+	Username string
+	Password string
+
+	Authorization string
+
+	Log telegraf.Logger `toml:"-"`
+
+	ResponseTimeout internal.Duration `toml:"response_timeout"`
+
+	tls.ClientConfig
+
+	client *http.Client
+}
+
+var sampleConfig = `
+  # Config for get statistics from JBoss AS
+  servers = [
+    "http://localhost:9090/management",
+  ]
+  
+  ## Username and password
+  username = ""
+  password = ""
+  
+  ## authorization mode could be "basic" or "digest"
+  authorization = "digest"
+
+  ## Optional SSL Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use SSL but skip chain & host verification
+  # insecure_skip_verify = false
+  ## Metric selection
+  metrics =[
+	"jvm",
+	"web", 		# Handles both EAP <=6.X/AS <=7.X and EAP >=7.X/Widlfly > 8
+	"deployment",
+	"database",
+	"transaction",
+	"jms",
+  ]
+`
+
+func (*JBoss) SampleConfig() string {
+	return sampleConfig
+}
+
+func (*JBoss) Description() string {
+	return "Telegraf plugin for gathering metrics from JBoss AS"
```

Comment:
Please leave out `Telegraf` here
```suggestion
	return "Plugin for gathering metrics from JBoss AS"
```